pymc3/distributions/multivariate.py

```diff
@@ -486,21 +486,43 @@ def __init__(self, n, p, *args, **kwargs):
 
         p = p / tt.sum(p, axis=-1, keepdims=True)
 
+        self._n = n
         if len(self.shape) == 2:
             try:
                 assert n.shape == (self.shape[0],)
             except AttributeError:
                 # this occurs when n is a scalar Python int or float
                 n *= tt.ones(self.shape[0])
-
             self.n = tt.shape_padright(n)
             self.p = p if p.ndim == 2 else tt.shape_padleft(p)
         else:
             self.n = tt.as_tensor_variable(n)
             self.p = tt.as_tensor_variable(p)
 
         self.mean = self.n * self.p
-        self.mode = tt.cast(tround(self.mean), 'int32')
+        self._raw_mode = tt.cast(tround(self.mean), 'int32')
+        if self.mean.ndim > 1:
+            self._n *= np.ones(self.shape[-2], dtype='int')
+
+        def calc_mode(raw_mode1d, mean1d, n):
+            n_missing = (n - raw_mode1d.sum())
+            to_increment = (mean1d - raw_mode1d).argsort()[-n_missing:]
+            return tt.switch(
+                tt.eq(n_missing, 0),
+                raw_mode1d,
+                tt.inc_subtensor(raw_mode1d[to_increment], 1)
+            )
+
+        try:
```

Comment:
why not just use `self.n` which is a tensor after line 500, so you don't need to check all these different cases?

Reply:
IIRC, I can't use a tensor to slice, so I actually need a non-tensor.  This is a problem when the n passed in to the constructor ***is*** a tensor (as in [test_multinomial](https://github.com/tsdlovell/pymc3/blob/82dabb5d90646d7176c12ec10c18a64c99c95551/pymc3/tests/test_distributions.py#L736)), which is why I had to add the except on line 524.

Reply:
Oh I see. Let me test it locally. The problem is there are use cases where `n` is a tensor (e.g., minibatch), so this approach might not work.